cloud-controller-manager/do/firewall_controller_print.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/digitalocean/godo"
+)
+
+func printRelevantFirewallParts(fw *godo.Firewall) string {
+	if fw == nil {
+		return "<n/a>"
+	}
+	return printFirewallParts(fw.Name, fw.InboundRules, fw.OutboundRules, fw.Tags)
+}
+
+func printRelevantFirewallRequestParts(fr *godo.FirewallRequest) string {
+	if fr == nil {
+		return "<n/a>"
+	}
+	return printFirewallParts(fr.Name, fr.InboundRules, fr.OutboundRules, fr.Tags)
+}
+
+func printFirewallParts(name string, inboundRules []godo.InboundRule, outboundRules []godo.OutboundRule, tags []string) string {
+	parts := []string{fmt.Sprintf("Name:%s", name)}
+	parts = append(parts, fmt.Sprintf("inRules:[%s]", printInboundRules(inboundRules)))
+	parts = append(parts, fmt.Sprintf("outRules:[%s]", printOutboundRules(outboundRules)))
+	parts = append(parts, fmt.Sprintf("Tags:%s", tags))
+
+	return strings.Join(parts, " ")
+}
+
+func printInboundRules(inboundRules []godo.InboundRule) string {
+	inbRules := make([]string, 0, len(inboundRules))
+	for _, inbRule := range inboundRules {
+		inbRules = append(inbRules, printInboundRule(inbRule))
+	}
+	sort.Strings(inbRules)
+	return strings.Join(inbRules, " ")
+}
+
+func printInboundRule(inboundRule godo.InboundRule) string {
+	portRange := inboundRule.PortRange
+	if inboundRule.PortRange == "" || inboundRule.PortRange == "all" {
+		portRange = "0"
+	}
+	rule := fmt.Sprintf("<Proto:%s PortRange:%s", inboundRule.Protocol, portRange)
+
+	if inboundRule.Sources != nil {
+		rule += fmt.Sprintf(" AddrSources:%s", inboundRule.Sources.Addresses)
+	}
+
+	rule += ">"
```

Comment:
nit: can we also make this a constant value. I'm assuming that this is the end/closing character for the rule output, but I'm not sure. Maybe we can add a descriptive constant or a comment that could make it more obvious. Wdyt?